pwd_guess_unit.py

```diff
@@ -220,6 +220,28 @@ def get_im(key):
         config.context_length = 4
         t = pwd_guess.TokenizingCharacterTable(config)
 
+    def test_encode_embedding(self):
```

Comment:
I think you're testing this in the wrong class here. I think you want to test it using CharacterTable and OptimizingCharacterTable and not TokenizingCharacterTable. It's great that it seems to work with the tokenizing table, but that use case isn't really useful. We really want to test the main use case that your experiments used, so that when we make changes to the code later, we're sure that those use cases still work. 

See CharacterTableTest and OptimizingTableTest. I think a better way of testing these is to add a new method to each of those to make sure that both use cases are tested.

Also, could you please check that the other methods of the character table classes work the way they should, I think the encode method has a bug, since it won't make the proper shape.